packages/hash/frontend/src/pages/[account-slug]/types/entity-type/entity-type-definition-tab.tsx

```diff
@@ -0,0 +1,45 @@
+import {
+  EntityType,
+  extractBaseUri,
+  VersionedUri,
+} from "@blockprotocol/type-system-web";
+import { Box, Typography } from "@mui/material";
+import { useFormContext, useWatch } from "react-hook-form";
+import { EntityTypeEditorForm } from "./form-types";
+import { TabButton } from "./tab-button";
+
+export const EntityTypeDefinitionTab = ({
+  entityType,
+}: {
+  entityType: EntityType;
+}) => {
+  const { control } = useFormContext<EntityTypeEditorForm>();
+  const propertiesCount = useWatch({ control, name: "properties.length" });
+
```

Comment:
What is the motivation behind not using the following, but patch the library to be able to use `properties.length` as a key?

```jsx
const properties = useWatch({ control, name: "properties" })
const propertiesCount = useMemo(() => properties.length, [properties])
```

Reply:
You can already use .length as a key in the library. It works. The typings are wrong, but I'm having some trouble convincing the author of that, so I've patched the typings. 

The motivation for watching the length and not the properties is it prevents re-render whenever anything in the form changes, and instead only re-renders when the number of properties changes.